addon/components/time-series-chart.js

```diff
@@ -210,6 +215,43 @@ const TimeSeriesChartComponent = ChartComponent.extend(LegendMixin,
     return new Date(time);
   },
 
+  // We'd like to have the option of turning our labels vertical when circumstances
+  // require.  This function gets ALL the labels of the xAxis and rotates them.
+  _rotateXAxisLabels: function() {
+    var gXAxis = this.get('xAxis');
+
+    // If we have a legend it'll take care of the margin bottom adjustments,
+    // else we need to give ourselves some more room for the labels.
+    if (!this.get('hasLegend')) {
+      this.set('marginBottom', 20);
+    }
+
+    gXAxis.selectAll('text')
+      .attr("y", 8)
+      .attr("x", -8)
+      .attr("dy", ".2em")
+      .attr("transform", "rotate(-60)")
+      .style("text-anchor", "end");
+
+    // we also need to mod the legend top padding
+    this.set('legendTopPadding', 30);
+  },
+
+  _shouldLabelsRotate: function () {
+    var labels, maxLabelWidth;
+
+    if (this.get('xAxisSmartLabels')) {
+      labels = this.get('xAxis').selectAll('text');
+      maxLabelWidth = this.get('maxLabelWidth') || this.get('_innerTickSpacingX');
+      labels.each(function () {
+        if (this.getBBox().width > maxLabelWidth) {
+          return true;
+        }
+      });
+    }
+    return false;
+  },
+
```

Comment:
I would consider making a couple of changes:

1. Testing the override property `xAxisVertLabels` in this function instead of making the function's single caller do it.
2. Moving the case with less code first, that is,`if ( ! this.get('xAxisSmartLabels') ) { return false; } /* else rest of code */` instead of the other way.